app/models/manageiq/providers/nuage/network_manager/event_catcher/stream.rb

```diff
@@ -62,6 +62,23 @@ def with_fallback_urls(urls)
     false
   end
 
+  # TODO: Don't ignore idle timeout error when following bug in qpid_proton is fixed:
+  # https://bugzilla.redhat.com/show_bug.cgi?id=1418026
+  # Until then, we don't want this error to keep restarting our event catcher in case there are no events emitted
+  # for 16 seconds.
+  def run_with_idle_timeout_ignored
+    loop do
+      begin
+        connection.run
+      rescue MiqException::MiqHostError => e
+        raise unless e.message.include?('local-idle-timeout expired')
+        $nuage_log.debug("#{self.class.log_prefix} Silently reconnect on 'local-idle-timeout expired'")
+        stop
```

Comment:
This effectively closes the connection on the [messaging handler](https://github.com/miha-plesko/manageiq-providers-nuage/blob/8f74cab92dfdae1614ea1f6b186bd0e9cc6fab70/app/models/manageiq/providers/nuage/network_manager/event_catcher/messaging_handler.rb#L46). Does this not occur automatically in case of an exception?